### PR TITLE
Cyclops adv class, the third times the charm (need help real bad please)

### PR DESCRIPTION
**Pull request by Hmeister-real**
## About The Pull Request

 There was an issue that made me remake the PR (i hope its fixed now). Also 20+ commit bloat again on #166 so thats another reason. If any checks fail il be pretty bummed out.

### CODE REVIEW

**Comment by Hmeister-real** [comment]
ok i have no clue why it's failing then, need help. Maybe i need to remake my fork or something?

**Comment by Hmeister-real** [comment]
> The repo that the container is trying to grab dependencies from is spitting errors for some reason, nothing to do with you. Hopefully it should self correct in time.

im not on my main PC rn, don't have Byond installed, maybe it's trying to check something from Byond but can't? Il make it Re-check on my main PC when i can, but it's probably safe to merge if it's unrelated to the actual PR

**Comment by CVSthePharmacy** [comment]
oh is this like, if you have the one eye quirk? it'd give access to a special adventurer class?

**Comment by Hmeister-real** [comment]
> oh is this like, if you have the one eye quirk? it'd give access to a special adventurer class?

exactly

**Comment by Hmeister-real** [comment]
shit i put the class into Rare rather than combat

**Comment by ChungusGamer666** [comment]
Does not compile. Provide proof of testing.

**Comment by Hmeister-real** [comment]
okay thank god, back to just 1 error from 96 before. trying to fix it, but i think the errors not important

**Comment by Hmeister-real** [comment]
i am literally going insane. I once again ask for some experienced person to help

**Comment by OrbitalParadise** [comment]
/mob/living/carbon/human/proc/advsetup()
	if(!advsetup)
		testing("RETARD")
		return TRUE
	var/blacklisted = check_blacklist(ckey)
	if(possibleclass.len && !blacklisted)
		var/datum/advclass/C = input(src, "What is my class?", "Adventure") as null|anything in sortNames(possibleclass)
		if(C && advsetup)
			if(C.maxchosen > -1)
				for(var/datum/advclass/A in GLOB.adv_classes)
					if(A.type == C.type)
						if(A.amtchosen >= A.maxchosen)
							possibleclass -= C
							to_chat(src, "<span class='warning'>Not enough slots for [C] left! Choose something different.</span>")
							return FALSE
						else
							A.amtchosen++
			if(alert(src, "[C.name]\n[C.tutorial]", "Are you sure?", "Yes", "No") != "Yes")
				return FALSE
			if(advsetup)
				advsetup = 0
				C.equipme(src)
				if(C.type == (/datum/advclass/rogue))
					src.grant_language(/datum/language/dwarvish)
				invisibility = 0
				cure_blind("advsetup")
				return TRUE
				
				
im not experienced in any way but I have been in your place, try doing if(C.type == (/datum/advclass/cyclops))
                                                                                                                         src.pre_equip(cyclops) 
 see if that works

**Comment by Hmeister-real** [comment]
> /mob/living/carbon/human/proc/advsetup() if(!advsetup) testing("RETARD") return TRUE var/blacklisted = check_blacklist(ckey) if(possibleclass.len && !blacklisted) var/datum/advclass/C = input(src, "What is my class?", "Adventure") as null|anything in sortNames(possibleclass) if(C && advsetup) if(C.maxchosen > -1) for(var/datum/advclass/A in GLOB.adv_classes) if(A.type == C.type) if(A.amtchosen >= A.maxchosen) possibleclass -= C to_chat(src, "Not enough slots for [C] left! Choose something different.") return FALSE else A.amtchosen++ if(alert(src, "[C.name]\n[C.tutorial]", "Are you sure?", "Yes", "No") != "Yes") return FALSE if(advsetup) advsetup = 0 C.equipme(src) if(C.type == (/datum/advclass/rogue)) src.grant_language(/datum/language/dwarvish) invisibility = 0 cure_blind("advsetup") return TRUE
> if(C.type == (/datum/advclass/cyclops)) src.pre_equip(cyclops) 

where would i put this, cyclops.dm?

**Comment by OrbitalParadise** [comment]
No that's not how it works, that's all in adventurer.dm you can insert src grant cyclops in adventurer dm then it calls your cyclops dm

File: code/modules/jobs/job_types/roguetown/adventurer/types/_advclass.dm [hunk 1]
```diff
@@ -11,6 +11,10 @@
 	"Dwarf",
 	"Dwarf"
 	)
+
+/datum/advclass/Initialize()
+	allowed_flaws = GLOB.character_flaws
```
Comment (BlueMemesauce) [inline code/modules/jobs/job_types/roguetown/adventurer/types/_advclass.dm:16]: you didn't initialize this variable (putting it under Initialize doesnt initialize the variable, which is confusing)

Comment (Hmeister-real) [reply code/modules/jobs/job_types/roguetown/adventurer/types/_advclass.dm:16]: so how would i actually initialise it? give it a value? or like, set it as initialised?

Comment (BlueMemesauce) [reply code/modules/jobs/job_types/roguetown/adventurer/types/_advclass.dm:16]: jut put it like the other vars and you have to put var/ in front of it to initialize it

File: code/modules/jobs/job_types/roguetown/adventurer/adventurer.dm [hunk 2]
```diff
@@ -217,3 +219,33 @@ GLOBAL_VAR_INIT(adventurer_hugbox_duration_still, 3 MINUTES)
...
+							return FALSE
+						else
+							A.amtchosen++
+	if(alert(src, "[C.name]\n[C.tutorial]", "Are you sure?", "Yes", "No") != "Yes")
+	return FALSE
+		if(advsetup)
+			advsetup = 0
+			C.equipme(src)
+		if(C.type == (/datum/advclass/rogue))
+			src.grant_language(/datum/language/dwarvish)
```
Comment (OrbitalParadise) [inline code/modules/jobs/job_types/roguetown/adventurer/adventurer.dm:246]: you gotta take out line 244 to 246

File: code/modules/jobs/job_types/roguetown/adventurer/adventurer.dm [hunk 3]
```diff
@@ -217,3 +219,33 @@ GLOBAL_VAR_INIT(adventurer_hugbox_duration_still, 3 MINUTES)
...
+			invisibility = 0
+			cure_blind("advsetup")
+			return TRUE
+	if(C.type == (/datum/advclass/cyclops))
+		src.pre_equip(cyclops)
```
Comment (OrbitalParadise) [inline code/modules/jobs/job_types/roguetown/adventurer/adventurer.dm:251]: does this work? might need to do src.pre_equip(/datum/advclass/cyclops)

**Comment by OrbitalParadise** [comment]
Are you testing this out in your own hosted dream maker server?

**Comment by BlueMemesauce** [comment]
yea if you want this pr to work you gotta start testing and using git on your computer
clearly making hundreds of web edits isn't working

**Comment by JTGSZ** [comment]
Closing
Talk with people in the channel/actually do something other than webedits. Yes?

**Comment by Hmeister-real** [comment]
remove help wanted @NGGJamie so this stops popping up on peoples home feed pls

**Pull request closed without merging**